### PR TITLE
Docker and more lightweight collector

### PR DESCRIPTION
**Pull request by vidma**
- [x] Add docker or scripts for building & running everything (Wildfly, MySQL, Zipkin)
- [x] Remove uneeded code & dependencies
- [x] update to use Zipkin reporter
- [x] update how process context is sent
- [x] add auth via Header
- [x] Move config to environment variables so jar could be precompiled
- [ ] Can we also remove jaeger-client now?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,14 +4,54 @@ An example of a Rest web application, using JAX-RS, Hibernate, MySQL, and Clickh
 
 The data lineages and data quality are tracked automatically between HTTP endpoints and Databases using Kensu collectors.
 
-By default the Jaeger backend tracer is used to keep the spans published to a observability platform, 
-this tracer is combined with the Kensu tracer to collect what is important for Kensu to build lineage and stats. 
+By default, the `Jaeger backend tracer + Zipkin reporter` is used to keep the spans published to a observability platform, this tracer is combined with the Kensu tracer to collect what is important for Kensu to build lineage and stats.
```
Comment (leszekgruchala) [inline README.md:7]: `to a` -> `to an`

File: src/main/java/io/kensu/collector/config/KensuTracerFactory.java [hunk 2]
```diff
@@ -32,19 +68,8 @@ public Tracer getTracer() {
...
-        } catch (Exception e) {
-            logger.warning("Can't create Jaeger as var env are not set, and we don't support default-tracer from Wildly yet");
-            backendTracer = NoopTracerFactory.create();
-        }
-
-        // Configure Kensu Tracer
-        System.setProperty("DAM_OFFLINE", properties.getProperty("kensu.collector.api.offline.switch"));
-        System.setProperty("DAM_OFFLINE_FILE_NAME", properties.getProperty("kensu.collector.api.offline.file"));
+        // FIXME: Need to send at least some of these values to Kensu ingestion
+        // FIXME: refactor...
```
Comment (leszekgruchala) [inline src/main/java/io/kensu/collector/config/KensuTracerFactory.java]: Do you want to merge FIXME and TODO to the OSS project?

File: src/main/java/io/kensu/collector/config/KensuTracerFactory.java [hunk 3]
```diff
@@ -53,17 +78,75 @@ public Tracer getTracer() {
...
+            String zipkinEndpointCfgKey = "ZIPKIN_ENDPOINT";
+            String kensuZipkinEndpoint = System.getProperty(zipkinEndpointCfgKey, System.getenv(zipkinEndpointCfgKey));
+            // FIXME: check, maybe jaeger is not needed at all anymore, and we could use Zipkin Tracer directly?
+            io.jaegertracing.spi.Reporter zipkinReporter = new io.jaegertracing.internal.reporters.CompositeReporter(
+                    buildZipkinReporter(
+                            kensuZipkinEndpoint,
+                            damEnv.damIngestionToken(),
+                            kensuAnnotations),
+                    buildZipkinReporter(
+                            "http://host.docker.internal:9411/api/v1/spans",
```
Comment (leszekgruchala) [inline src/main/java/io/kensu/collector/config/KensuTracerFactory.java]: Shouldn't it be taken from the configuration?

File: src/main/java/io/kensu/collector/config/KensuTracerFactory.java [hunk 4]
```diff
@@ -53,17 +78,75 @@ public Tracer getTracer() {
...
+                    return conn;
+                }
+            });
+        }
+        // P.S. there's no non-thrift span converter. originally it uses String.valueOf for tag values...!
+        RemoteReporter reporter =  new RemoteReporter.Builder()
+                .withSender(ZipkinSender.create(
+                        URLConnectionSender.newBuilder().encoding(Encoding.THRIFT).endpoint(serverUrl).build()
+                ))
+                .withFlushInterval(1) // FIXME
```
Comment (leszekgruchala) [inline src/main/java/io/kensu/collector/config/KensuTracerFactory.java:145]: Please provide at least info what's here to fix

File: src/main/java/io/kensu/collector/config/DamProcessEnvironment.java [hunk 5]
```diff
@@ -1,29 +1,18 @@
...
-        return serverHost;
-    }
+//
+//    public String damIngestionUrl(){
+//        String serverHost = getOptEnv("DAM_INGESTION_URL", null);
+//        if (serverHost == null) {
+//            throw new RuntimeException("DAM_INGESTION_URL env var / java prop required when DAM is in online mode, e.g. https://localhost");
+//        }
+//        return serverHost;
+//    }
```
Comment (leszekgruchala) [inline src/main/java/io/kensu/collector/config/DamProcessEnvironment.java]: Why is this commented out and left here?

File: src/main/java/io/kensu/collector/config/DamProcessEnvironment.java [hunk 6]
```diff
@@ -33,78 +22,45 @@ public String damIngestionToken(){
...
+//        if (projectsStr != null){
+//            List<String> projects = Arrays.stream(projectsStr.trim()
+//                                            .split(";")).map(String::trim)
+//                                            .filter(e->e.length()>0)
+//                                            .collect(Collectors.toList());
+//            if (projects.size() == 0) return null;
+//            return projects;
+//        } else {
+//            return null;
+//        }
```
Comment (leszekgruchala) [inline src/main/java/io/kensu/collector/config/DamProcessEnvironment.java]: comment?

File: src/main/java/io/kensu/collector/config/KensuTracerFactory.java (hunk 2 above)
Comment (leszekgruchala) [inline src/main/java/io/kensu/collector/config/KensuTracerFactory.java]: Do you want to keep those comments for OSS?

**Review by leszekgruchala — COMMENTED** [review]
Not sure about all those FIXMEs and commented code. Do you want to push that to OSS?

File: src/main/java/io/kensu/collector/config/DamProcessEnvironment.java (hunk 6 above)
Comment (vidma) [reply src/main/java/io/kensu/collector/config/DamProcessEnvironment.java]: good catch. indeed here the idea was to further think if it's a good idea to send string like"project1;project2" without parsing the string (e.g. json would be a more standard format), but keep things simple for simplicity it's probably fine?

**Pull request merged**